### DISCUSSION TITLE
Support for environments in secrets for Dependabot

### DISCUSSION THREAD

**Discussion by ThisIsMissEm** [community/community#40090] [Actions]
### Select Topic Area

Product Feedback

### Body

Currently dependabot has the ability to have a set of repository and organisation secrets, however, it lacks the ability to access the Environments secrets, this means test matrixes that use Environments cannot be run by dependabot PRs.

It'd be fantastic if we could add Environments that Dependabot has access to.

### Related

https://github.com/community/community/discussions/27505

**Comment by Th3S4mur41** [comment]
Same issue but even going further than the PR itself...
If dependabot merges a PR, the secret source for the subsequent push workflow is not actions, but rather dependabot.

In this repo, pushing to main is setup to trigger a semantic release which can publish a new package using NPM_TOKEN set in environment secrets.

[The release fails](https://github.com/Th3S4mur41/release-config/issues/38) when the push workflow is triggered by [dependabot merging one of its PR](https://github.com/Th3S4mur41/release-config/pull/37)

![image](https://user-images.githubusercontent.com/5543595/210546402-7f9807b0-b778-4959-8269-5eb4243df268.png)

[The same workflow triggered by a manual push or a PR merged by a user](https://github.com/Th3S4mur41/release-config/actions/runs/3714349227/jobs/6298173476) works fine, since it has a different secret source and can therefor access the environment
![image](https://user-images.githubusercontent.com/5543595/210547045-cbca3417-7e98-4753-99fe-ef2fe0f9a0ef.png)

In this example, since there is only one environment, this is just annoying because we have to set the NPM_TOKEN both in the production environment and the dependabot secrets.
But that would break the workflow for workflows with multiple environments deploying with different tokens (e.g. staging + production).
